mysql-test/suite/rocksdb/t/ttl_primary.test

```diff
@@ -0,0 +1,498 @@
+set global rocksdb_compact_cf='default';
+
+# should have filtered the rows out since ttl is passed in compaction filter
+SELECT COUNT(*) FROM t1;
+DROP TABLE t1;
+
+# multi-part PK w/ TTL
+CREATE TABLE t1 (
+  a bigint(20) NOT NULL,
+  b int NOT NULL,
+  c int NOT NULL,
+  ts bigint(20) UNSIGNED NOT NULL,
+  PRIMARY KEY (a,c)
+) ENGINE=ROCKSDB
+COMMENT='ttl_duration=1;ttl_col=ts;';
+
+INSERT INTO t1 values (1, 3, 5, UNIX_TIMESTAMP());
+INSERT INTO t1 values (2, 4, 6, UNIX_TIMESTAMP());
+SELECT COUNT(*) FROM t1;
+
+--sleep 1
+set global rocksdb_force_flush_memtable_now=1;
+set global rocksdb_compact_cf='default';
+
+# should have filtered the rows out since ttl is passed in compaction filter
+SELECT COUNT(*) FROM t1;
+DROP TABLE t1;
+
+# nullable column(s) before TTL
+CREATE TABLE t1 (
+  a bigint(20) NOT NULL,
+  b int,
+  c int,
+  ts bigint(20) UNSIGNED NOT NULL,
+  PRIMARY KEY (a)
+) ENGINE=ROCKSDB
+COMMENT='ttl_duration=1;ttl_col=ts;';
+
+INSERT INTO t1 values (1, NULL, NULL, UNIX_TIMESTAMP());
+INSERT INTO t1 values (2, NULL, NULL, UNIX_TIMESTAMP());
+SELECT COUNT(*) FROM t1;
+
+--sleep 1
+set global rocksdb_force_flush_memtable_now=1;
+set global rocksdb_compact_cf='default';
+
+# should have filtered the rows out since ttl is passed in compaction filter
+SELECT COUNT(*) FROM t1;
+DROP TABLE t1;
+
+# variable len columns + null column(s) before TTL
+CREATE TABLE t1 (
+`a` binary(8) NOT NULL,
+`b` varbinary(64),
+`c` varbinary(256),
+`ts` bigint(20) UNSIGNED NOT NULL,
+`value` mediumblob NOT NULL,
+PRIMARY KEY (`a`)
+) ENGINE=ROCKSDB DEFAULT CHARSET=latin1
+COMMENT='ttl_duration=1;ttl_col=ts;';
+
+INSERT INTO t1 values ('a', NULL, 'bc', UNIX_TIMESTAMP(), 'd');
+INSERT INTO t1 values ('d', 'efghijk', NULL, UNIX_TIMESTAMP(), 'l');
+SELECT COUNT(*) FROM t1;
+
+--sleep 1
+set global rocksdb_force_flush_memtable_now=1;
+set global rocksdb_compact_cf='default';
+
+# should have filtered the rows out since ttl is passed in compaction filter
+SELECT COUNT(*) FROM t1;
+DROP TABLE t1;
+
+# TTL implicitly generated (no ttl column)
+CREATE TABLE t1 (
+  a bigint(20) NOT NULL,
+  b int NOT NULL,
+  c int NOT NULL,
+  PRIMARY KEY (a)
+) ENGINE=ROCKSDB
+COMMENT='ttl_duration=1;';
+
+INSERT INTO t1 values (1, 3, 5);
+INSERT INTO t1 values (2, 4, 6);
+SELECT COUNT(*) FROM t1;
+
+--sleep 1
+set global rocksdb_force_flush_memtable_now=1;
+set global rocksdb_compact_cf='default';
+
+# should have filtered the rows out since ttl is passed in compaction filter
+SELECT COUNT(*) FROM t1;
+DROP TABLE t1;
+
+# TTL field as the PK
+CREATE TABLE t1 (
+  ts bigint(20) UNSIGNED NOT NULL,
+  PRIMARY KEY (ts)
+) ENGINE=ROCKSDB
+COMMENT='ttl_duration=5;ttl_col=ts;';
+
+INSERT INTO t1 values (UNIX_TIMESTAMP());
+--sleep 1
+INSERT INTO t1 values (UNIX_TIMESTAMP());
+SELECT COUNT(*) FROM t1;
+
+set global rocksdb_force_flush_memtable_now=1;
+set global rocksdb_compact_cf='default';
+# should all still be there..
+SELECT COUNT(*) FROM t1;
```

Comment:
True. There are two commits coming from upstream, one in next batch (May) and one in the following (June) that work towards addressing flakyness:
https://github.com/facebook/mysql-5.6/commit/ad90bf77f3a353336c0e33f5784eb9b30039880a
https://github.com/facebook/mysql-5.6/commit/891a9115e3d16e1a2b2efab11eff1880eb69ce1e

I would prefer to keep these aligned with upstream for now until I get through the merges and if we still have stability issues, then we can work through fixing them and sending the changes upstream.